Update APIs to ret ChInfo on open/update/close ch

manoranjith opened the pull request:
<!-- Provide a general summary of your changes in the title above

Please read our contribution guidelines and sign the Contributor License
Agreement (CLA) before submitting the pull request. Also, check if there are no
other open pull requests targeting the same issue. -->
#### Description
<!-- Describe your changes in detail. -->

 - This is changeset 4 (final) for using uniform data formats accross different
 APIs.
    
 - Update RespondChProposal, SendChUpdate, RespondChUpdate and CloseCh
 APIs to return ChInfo on successful execution. Update ch.GetBalInfo also
 to return ChInfo and rename it to ch.GetChInfo.
    
- In [app/payment] SendPayChUpdate, two mutex protected calls were used
in succession:To fix this, add methods to read fixed parameters of a channel without mutex
protection and use it read Currency and Parts.

##### Category
<!-- Tell us what type of issue does your pull request target.
You can uncomment one of the following options: -->

<!-- Bug Fix -->
<!-- Improvement -->
Implementation Task

##### Relevant issue
<!-- Provide a link to the related issue. You can use the following keywords
and the issue number: "fixes", "resolves", "relates to". E.g.: closes #21

We accept only pull requests related to open issues. If you're suggesting a new
feature, improvement or fixing a bug that is not yet reported, please discuss it in
an issue before submitting a pull request. -->

Closes #108.

#### Testing
<!-- Tell us how you have tested the changes. -->

No new functionality / tests were added, all existing tests should pass.
One test 

##### Steps to run the tests
<!-- Describe a set of steps to run the tests relevant to this change. -->

#### Checklist 
<!-- Please check if the pull request fulfils these requirements: -->

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

manoranjith commented:
This should be merged only after #115.

anagha-ks commented on app/payment/channel_test.go:
```diff
@@ -163,25 +169,26 @@ func Test_RespondPayChUpdate(t *testing.T) {
+		chAPI.On("RespondChUpdate", context.Background(), updateID, accept).Return(updatedChInfo, nil)
 
-		gotErr := payment.RespondPayChUpdate(context.Background(), chAPI, updateID, accept)
+		gotPayChInfo, gotErr := payment.RespondPayChUpdate(context.Background(), chAPI, updateID, accept)
```
Here, since `RespondPayChUpdate` is giving `updatedPayChInfo`, may be we can use the term like `updatedPayChInfo` so that we can understand from the name itself.

manoranjith replied:
Updated in [this commit](https://github.com/hyperledger-labs/perun-node/pull/116/commits/2f759e50555fbcb313b10428ddd4ab7ff753604a).

anagha-ks commented on app/payment/session_test.go:
```diff
@@ -207,25 +207,26 @@ func Test_RespondPayChProposal(t *testing.T) {
+		sessionAPI.On("RespondChProposal", context.Background(), proposalID, accept).Return(openedChInfo, nil)
 
-		gotErr := payment.RespondPayChProposal(context.Background(), sessionAPI, proposalID, accept)
+		gotPayChInfo, gotErr := payment.RespondPayChProposal(context.Background(), sessionAPI, proposalID, accept)
```
Here also something like `payChannelProposalInfo` variable to store `RespondPayChProposal()`output

manoranjith replied:
Updated in [this commit](https://github.com/hyperledger-labs/perun-node/pull/116/commits/2f759e50555fbcb313b10428ddd4ab7ff753604a).

Commit:
Update APIs to ret ChInfo on open/update/close ch

- This is changeset 4 (final) for using uniform data formats accross different
APIs.

- Update RespondChProposal, SendChUpdate, RespondChUpdate and CloseCh
APIs to return ChInfo on successful execution. Update ch.GetBalInfo also
to return ChInfo and rename it to ch.GetChInfo.

- In [app/payment] SendPayChUpdate, two mutex protected calls were used
in succession: GetChInfo and SendChUpdate and variable data pertaining
the channel read from the first calls was used in the second. This could
lead to errors, if data has changed in the intermitent duration. To fix
this, add methods to read fixed parameters of a channel without mutex
protection and use it read Currency and Parts. Also move the balance
manipulation function into the updater.

Closes #108.

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>

Commit:
[review commit] rename variable

Signed-off-by: Manoranjith <ponraj.manoranjitha@in.bosch.com>
